elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryThread.java

```diff
@@ -150,93 +146,80 @@ else if (queryObj.getQueryType().equals(QueryType.GRAPHQL_V1_0)) {
      * @return rec is the recordCount
      */
     protected Integer calculateRecordsJSON(String jsonStr) {
-        Integer rec;
-        try {
-            JSONObject j = new JSONObject(jsonStr);
-            rec = j.getJSONArray("data").length();
-
-        } catch (JSONException e) {
-            rec = null;
+        Integer rec = null;
+        JsonElement jsonTree = JsonParser.parseString(jsonStr);
+        if (!jsonTree.isJsonObject()) {
+            return null;
+        }
+        JsonObject jsonObject = jsonTree.getAsJsonObject();
+        JsonElement jsonElement = jsonObject.get("data");
+        if (jsonElement.isJsonArray()) {
+            rec = jsonElement.getAsJsonArray().size();
         }
+
         return rec;
     }
 
     /**
      * This method calculates the number of records from the response with GRAPHQL API.
-     * @param response is the response.getBody() we get from the response
-     * @param table_name is the table from which we extract the data
+     * @param jsonStr is the response.getBody() we get from the response
      * @return rec is the recordCount
      */
-    protected Integer calculateRecordsGraphQL(String response, String table_name) {
-        Integer rec;
+    protected Integer calculateRecordsGraphQL(String jsonStr) {
+        Integer rec = null;
+
         try {
-            JSONObject j = new JSONObject(response);
-            JSONObject j2 = j.getJSONObject("data");
-            JSONObject j3 = j2.getJSONObject(table_name);
-            rec = j3.getJSONArray("edges").length();
+            JsonReader jsonReader = new JsonReader(new StringReader(jsonStr));
+            JsonElement jsonTree = JsonParser.parseString(jsonStr);
+            if (!jsonTree.isJsonObject()) {
+                return null;
+            }
+            JsonObject jsonObject = jsonTree.getAsJsonObject();
+            while (jsonReader.hasNext()) {
+                JsonToken nextToken = jsonReader.peek();
 
-        } catch (JSONException e) {
-            rec = null;
-        }
-        return rec;
-    }
+                if (JsonToken.BEGIN_OBJECT.equals(nextToken)) {
 
-    /**
-     * This method helps to extract the table name from the query.
-     * @param jsonStr is the query with which we are extracting the data
-     * @return the table name from the above query
-     */
-    protected String getTableNameFromQuery(String jsonStr) {
-        StringBuilder str = new StringBuilder();
-        try {
-            JSONObject j = new JSONObject(jsonStr);
-            String s = (String) j.get("query");
-
-            Integer countBrackets = 0;
-            for (int i = 0; i < s.length(); i++) {
-                if (s.charAt(i) == '{' || s.charAt(i) == '}') {
-                    countBrackets++;
-                    if (countBrackets == 2) {
+                    jsonReader.beginObject();
```

Comment:
I wonder if this would be simpler with JsonPath using a wildcard for the top node.